### DISCUSSION TITLE
Possible 8BitDo GC bug + code question

### DISCUSSION THREAD

**Discussion by davistroy40** [darthcloud/BlueRetro#76] [General]
Hey Darth,

I just put together a quick rats nest version of BlueRetro for gamecube. So far so good as far as wiring and install goes! This is awesome!

However, I think I either found a bug or I am doing something wrong regarding 8bitdo controllers. In particular the Sn30 Pro+ v1 I have been testing with. I am unaware if this is an issue with Xbox or PS controllers since I do not own those.

 I played two games, Mario Sunshine and Double Dash, both of which use the analog triggers and full press discrete shoulder button for important gameplay features. The analog trigger from the Sn30 Pro works in both cases but there is no response for the full press. I am aware that most modern controller do not include the discrete button at a full trigger pull. Is it possible a filter of some sort could be added to emulated the discrete (something like if 90% pressed then send discrete signal)? I played around with their ultimate software (https://support.8bitdo.com/ultimate/sn30pro_plus.html) to no quick solution. Wondering if you have any ideas as to what could be causing this?

Also, for some odd reason recently, I have became obsessed with the GC-GBA link cable and have been thinking about making a bluetooth transmitter for link cable comms. I have done very little review of your code and how the controller inputs are handled, but do you think it could be possible for a GBA link cable to be support over something like Xinput? To my knowledge, its just serial data so it should not be that hard to send over bluetooth (and fork off of BlueRetro).

**Comment by darthcloud** [comment]
You need to use the GameCube merge trigger preset config. To map the end button at the of the analog one. See note at end of GameCube instructions: https://github.com/darthcloud/BlueRetro/wiki/BlueRetro-System-Specific-User-Manual#mapping-14

For GBA im not sure to understand question. Yes you could emulate such connection but then you would likely need explicit support for each games as they upload their own code on the GBA and likely use their own special protocol pass that point.

**Comment by davistroy40** [comment]
Oh thanks for pointing that out. My eyes must of totally skipped that! I'll try my best to upload the config this weekend.

Sounds like the GBA idea might be a bigger pain than I initially thought. Might be worth while to do some research if I ever decide to tackle this.

Thanks for the info!
